scripts/split.py

```diff
@@ -0,0 +1,53 @@
+import os
+import cv2
+
+
+def video_to_frames(input_file, output_path):
+    """
+    Extracts the frames from an input video file
+    and saves them as separate frames in an output directory.
+
+    Source: CS 445 Project 5 Utils
+
+    Input:
+        input_file: Input video file.
+        output_path: Output directorys.
+    Output:
+        None
+    """
+
+    cap = cv2.VideoCapture()
+    cap.open(input_file)
+
+    if not cap.isOpened():
+        print("Failed to open input video")
+
+    frame_count = cap.get(cv2.CAP_PROP_FRAME_COUNT)
+
+    frame_idx = 0
+    while frame_idx < frame_count:
+        ret, frame = cap.read()
+
+        if not ret:
+            print("Failed to get the frame {}".format(frame_idx))
+            return
+
+        out_name = os.path.join(output_path, "f{:04d}.jpg".format(frame_idx + 1))
+        ret = cv2.imwrite(out_name, frame)
+
+        if not ret:
+            print("Failed to write the frame {}".format(frame_idx))
+            return
+
+        frame_idx += 1
+        cap.set(cv2.CAP_PROP_POS_FRAMES, frame_idx)
+
+
+if __name__ == "__main__":
+    from sys import argv
+
+    input_video = argv[1]
+    name = input_video.split(".")[0]  # {name}.video
+    os.mkdir(name)
```

Comment:
Maybe change to `mkdirs`? It would be nice to be able to call this script from root and use arg "samples/sample.video". Then the created directory can be "samples/sample/". This way, the images are still ignored by git. I think you need `mkdirs` for recursive but I might be mistaken.

Reply:
But also maybe there's some argument for why we shouldn't allow this idk

Reply:
O I tested this calling it from root `python scripts/split.py samples/example.video` and it created a directory of frames in `samples/example`, but we could reorganize the structure too if you think it would be easier.

Reply:
whoops ok for some reason I thought mkdir doesnt work recursively